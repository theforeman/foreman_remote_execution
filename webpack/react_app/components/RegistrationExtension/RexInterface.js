import React from 'react';
import PropTypes from 'prop-types';

import { translate as __ } from 'foremanReact/common/I18n';

import { FormGroup, TextInput, Popover } from '@patternfly/react-core';

import { HelpIcon } from '@patternfly/react-icons';

const RexInterface = ({ isLoading, onChange }) => (
  <FormGroup
    label={__('Remote Execution Interface')}
    fieldId="reg_rex_interface"
    labelIcon={
      <Popover
        bodyContent={
          <div>
            {__('Identifier of the Host interface for Remote execution')}
          </div>
        }
      >
        <button
          className="pf-c-form__group-label-help"
          onClick={e => e.preventDefault()}
        >
          <HelpIcon noVerticalAlign />
        </button>
      </Popover>
    }
  >
    <TextInput
      type="text"
      onBlur={e => onChange({ remoteExecutionInterface: e.target.value })}
      id="reg_rex_interface_input"
      isDisabled={isLoading}
    />
  </FormGroup>
);

RexInterface.propTypes = {
  onChange: PropTypes.func,
  isLoading: PropTypes.bool,
};

RexInterface.defaultProps = {
  onChange: undefined,
  isLoading: false,
};

export default RexInterface;
