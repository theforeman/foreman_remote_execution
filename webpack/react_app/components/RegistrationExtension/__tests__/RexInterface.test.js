import { testComponentSnapshotsWithFixtures } from '@theforeman/test';
import RexInterface from '../RexInterface';

const fixtures = {
  renders: { isLoading: false, onChange: () => {} },
};

describe('RexInterface', () =>
  testComponentSnapshotsWithFixtures(RexInterface, fixtures));
