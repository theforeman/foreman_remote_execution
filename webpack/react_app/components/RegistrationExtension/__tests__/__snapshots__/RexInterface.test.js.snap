// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RexInterface renders 1`] = `
<FormGroup
  fieldId="reg_rex_interface"
  label="Remote Execution Interface"
  labelIcon={
    <Popover
      bodyContent={
        <div>
          Identifier of the Host interface for Remote execution
        </div>
      }
    >
      <button
        className="pf-c-form__group-label-help"
        onClick={[Function]}
      >
        <HelpIcon
          color="currentColor"
          noVerticalAlign={true}
          size="sm"
        />
      </button>
    </Popover>
  }
>
  <TextInput
    id="reg_rex_interface_input"
    isDisabled={false}
    onBlur={[Function]}
    type="text"
  />
</FormGroup>
`;
